From 64499ae4ccc312ac40bea251ed8530397322ed58 Mon Sep 17 00:00:00 2001
From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Sat, 8 Apr 2023 01:48:17 +0200
Subject: [PATCH] PPD generator for driverless printers: add
 "*LandscapeOrientation:"

In the generated PPD files add the keyword "*LandscapeOrientation:"
with the values "Plus90" or "Minus90" according to the
landscape-orientation-requested-preferred printer IPP attribute with
values "4" and "5" resp.
diff --git a/cupsfilters/ppdgenerator.c b/cupsfilters/ppdgenerator.c
index 0d4dd7097..7886d78ca 100644
--- a/cupsfilters/ppdgenerator.c
+++ b/cupsfilters/ppdgenerator.c
@@ -1812,6 +1812,16 @@ ppdCreateFromIPP2(char         *buffer,          /* I - Filename buffer */
       cupsFilePuts(fp, "*ColorDevice: False\n");
   }
 
+  if ((attr = ippFindAttribute(response,
+			       "landscape-orientation-requested-preferred",
+			       IPP_TAG_ZERO)) != NULL)
+  {
+    if (ippGetInteger(attr, 0) == 5)
+      cupsFilePuts(fp, "*LandscapeOrientation: Minus90\n");
+    else
+      cupsFilePuts(fp, "*LandscapeOrientation: Plus90\n");
+  }
+
   cupsFilePrintf(fp, "*cupsVersion: %d.%d\n", CUPS_VERSION_MAJOR,
 		 CUPS_VERSION_MINOR);
   cupsFilePuts(fp, "*cupsSNMPSupplies: False\n");
